larq_compute_engine/core/bgemm_impl_ruy.h

```diff
@@ -99,41 +97,16 @@ struct BGemmImplUsingRuy {
       bgemm_runtime_path = ruy::Path::kStandardCpp;
     }
 
-    ruy::Matrix<LhsScalar> transposed_lhs(lhs);
+    ruy::Mat<LhsScalar> transposed_lhs(internal_lhs);
     Transpose(&transposed_lhs);
 
     ruy::TrMulParams binary_trmul_params;
     CreateBinaryTrMulParams<BGemmCompiledPaths>(
-        transposed_lhs, rhs, mul_params, ruy_context, &dst, bgemm_runtime_path,
-        &binary_trmul_params);
-
-    // pre-pack the lhs and rhs matrices
-    ruy::PrepackedMatrix prepacked_lhs;
-    ruy::PrepackedMatrix prepacked_rhs;
-    ruy::SidePair<ruy::PrepackedMatrix*> prepacked(&prepacked_lhs,
-                                                   &prepacked_rhs);
-
-    const ruy::SidePair<int> origin{0, 0};
-    const ruy::SidePair<int> rounded_dims{
-        binary_trmul_params.packed[ruy::Side::kLhs].layout.cols,
-        binary_trmul_params.packed[ruy::Side::kRhs].layout.cols};
-
-    ruy::Tuning tuning = ContextInternal::GetMainThreadTuning(ruy_context);
-    for (ruy::Side side : {ruy::Side::kLhs, ruy::Side::kRhs}) {
-      if (prepacked[side]) {
-        prepacked[side]->data_size = DataSize(binary_trmul_params.packed[side]);
-        prepacked[side]->sums_size = SumsSize(binary_trmul_params.packed[side]);
-        prepacked[side]->data = alloc_fn(prepacked[side]->data_size);
-        prepacked[side]->sums = alloc_fn(prepacked[side]->sums_size);
-        binary_trmul_params.packed[side].data = prepacked[side]->data;
-        binary_trmul_params.packed[side].sums = prepacked[side]->sums;
-        binary_trmul_params.RunPack(side, tuning, origin[side],
-                                    rounded_dims[side]);
-        binary_trmul_params.is_prepacked[side] = true;
-      }
-    }
+        transposed_lhs, internal_rhs, mul_params, ruy_ctx, &internal_dst,
+        bgemm_runtime_path, &binary_trmul_params);
 
-    ruy::TrMul(&binary_trmul_params, ruy_context);
+    HandlePrepackedCaching(&binary_trmul_params, ruy_ctx);
+    ruy::TrMul(&binary_trmul_params, ruy_ctx);
```

Comment:
⚠️ Review with care. I am not familiar with this code so I am not sure what these changes do, but for now at least they seem to compile.

/cc @Tombana @arashb

Reply:
here I was using the RUY's advanced API but it is completely removed in [this](https://github.com/google/ruy/commit/57e64b4c8f32e813ce46eb495d13ee301826e498) commit. I need to dedicate more time to figure out how to replicate that.